src/main/lrsql/util/reaction.clj

```diff
@@ -101,3 +102,19 @@
   [ruleset-bytes]
   (with-open [r (io/reader ruleset-bytes)]
     (cjson/parse-stream r (partial keyword nil))))
+
+(s/fdef generate-statement
+  :args (s/cat :cond-map (s/map-of ::rs/condition-name
+                         ::rs/val)
+               :statement map?))
```

Comment:
In this case, we're really talking about a template, so the `s/cat` key should probably be `:template` (I see it is named that down in the function so this should match). As for the spec we _can't_ rely on it being a valid xAPI statement like we can above, because of the additional structures for paths so `map?` is OK, but we can be a *little* more specific by using `:xapi-schema.spec/any-json`.

Reply:
Also we're missing a `:ret` spec here, which would be `:xapi-schema.spec/statement` I believe.